logging_utilities/filters/django_request.py

```diff
@@ -21,16 +21,20 @@ def _pattern(text):
 class JsonDjangoRequest(logging.Filter):
     """Convert Django request to a json object
 
-    The django framework adds sometimes a request to the logs extra (HttpRequest or socket object).
-    This filter recursively converts this request object to a python dictionary that can be dumped
-    into a json string.
-    This is useful for example if you want to use this request with the JSON formatter.
+    This filter recursively converts a django HttpRequest object to a python dictionary that can be
+    dumped into a json string. This is useful for example if you want to log an extra parameter of
+    type HttpRequest with the JSON formatter. If the specified attribute is not of type HttpRequest,
+    it will simply be ignored and passed though.
 
     Additionally the attributes of the request that needs to be jsonify can be configured using the
     `include_keys` and/or `exclude_keys` parameters.
+
+    The django framework adds already sometimes a request to the logs extra (HttpRequest or socket
+    object). So if you use the default attribute name "request" for this filter, beware that you
+    will need to handle the case where the attribute is of type 'socket' separately.
     """
 
-    def __init__(self, include_keys=None, exclude_keys=None):
+    def __init__(self, include_keys=None, exclude_keys=None, attr_name='request'):
```

Comment:
For backward compatibility it is ok to use `request` as default. However due to the issue with the socket mentioned above this is not a good default choice. For use, as this filter is only used in one place, I would suggest to change the default to `http_request`, make a major version bump (backward compatibility is broken), make a note in readme about breaking changes, and set `attr_name=request` in service-stac.

Reply:
It's true that this would plead in favor of changing the name. However, django also logs requests of type "HttpRequest", and this could plead in favor of keeping the default name "request", as many people would probably want to log them.

Reply:
I know about django logging HttpRequest but I would use default value that works out of the bug, and we know that `request` will not work out of the box due to django logging it using http request and socket. Usually user are more interested on their logs as the logs of libraries.